### PR TITLE
#145540771/authenication-service-api

### PR DESCRIPTION
**Pull request by Seyi-Keye**
What does this PR do?
authentication Service Api

Description of Task to be completed?
A user should be able to call this service with email and password. and get a response of successful, invalid details or an error.

How should this be manually tested?
A user should be able to run `npm test` to test the component as well as pass email and password to the service.

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?

N/A

Screenshots (if appropriate)

N/A

Questions:

N/A

### CODE REVIEW

File: public/localforage.js [hunk 1]
```diff
@@ -0,0 +1,2313 @@
+/*!
```
Comment (abiodun0) [inline public/localforage.js:1]: Can this be in a library or something

File: src/api/Authentication.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    if(validEmail && validPassword) {
+      localforage.setItem('authenication', 'Successful');
+    }
+    return 'Successful';
+  }
+
+  static passwordChecker(password){
+    const trimedPassword = password.trim();
+    if(!trimedPassword.length) return 'Password cannot be empty';
+    else if(trimedPassword.length === 1 || password.length ===2) return 'Invalid password, length of 3 at least';
```
Comment (abiodun0) [inline src/api/Authentication.js:16]: ***Code Quality: Nitpick***
This looks good but it would be neater if there is no else statement at all
so this could still work like this
```if(trimedPassword.length === 1 || password.length ===2) return 'Invalid password, length of 3 at least'```

File: src/test/api/Authentication.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { expect } from 'chai';
+import { Authentication } from '../../api/Authentication';
+
+describe('Authentication Service Api' , () => {
+  it('returns successful on valid email and password', () => {
+    expect(Authentication.validityChecker('abc@gmail.com', 'abc')).to.equal('Successful');
```
Comment (abiodun0) [inline src/test/api/Authentication.test.js:6]: ***Code Quality: Awesome/Nitpick***
Excellent tests here. are we not suppose to test that localforage.setItem is called by spying on the setItem property?

File: src/test/api/Authentication.test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  });
+
+  it('returns error on invalid email', () => {
+    expect(Authentication.validityChecker('abc@gmailcom', 'abc')).to.equal('Invalid');
+  });
+});
+
+describe('Authentication email checker', () => {
```
Comment (abiodun0) [inline src/test/api/Authentication.test.js:14]: ***Code Quality: Awesome****
Awesome work here!

File: src/api/Authentication.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+export class Authentication{
+  constructor(){}
```
Comment (abiodun0) [inline src/api/Authentication.js:2]: ***Code Qaulity: Nitpick***
Since all we are doing is is static methods here this constructors are not needed at all here..

File: src/api/Authentication.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  static validityChecker(email, password){
+    const validPassword = Authentication.passwordChecker(password);
+    const validEmail = Authentication.emailChecker(email);
+    if(validEmail && validPassword) {
+      localforage.setItem('authenication', 'Successful');
+    }
+    return 'Successful';
```
Comment (abiodun0) [inline src/api/Authentication.js:10]: ***Code Quality: Nitpick***
Is this not suppose to return error if the above if condition fails?

**Review by abiodun0 — CHANGES_REQUESTED** [review]
Good work Seyi. Just some concerns about co-inciseness and taking care of edge cases. Keep this up!

File: src/test/api/Authentication.test.js (hunk 3 above)
Comment (Seyi-Keye) [reply src/test/api/Authentication.test.js:6]: I will test for that as well, just wanted to clarify if I was on track with the requirement.

File: public/localforage.js (hunk 1 above)
Comment (Seyi-Keye) [reply public/localforage.js:1]: This was the only way I could use the file
